### PR TITLE
Fat 5616 create personal account

### PR DESCRIPTION
**Pull request by Activict**

### CODE REVIEW

File: StoreApp.BLL/Services/UserValidateService.cs [hunk 1]
```diff
@@ -41,5 +45,20 @@ public string GetRole(UserDTO userDTO)
 
             return users.FirstOrDefault(u => u.UserName.Equals(userDTO.UserName) && u.Password.Equals(userDTO.Password))?.Role;
         }
+
+        public bool CheckTruePassword(UserDTO userDTO)
+        {
+            return DataBase.Users.GetAll()
+                                 .Any(u => u.Id.Equals(userDTO.Id) &&
+                                           u.Password.Equals(userDTO.Password));
```
Comment (AleksandroBoko) [inline StoreApp.BLL/Services/UserValidateService.cs]: Just apply default formatting

File: StoreApp.BLL/Services/UserValidateService.cs [hunk 2]
```diff
@@ -41,5 +45,20 @@ public string GetRole(UserDTO userDTO)
...
+        }
+
+        public bool CheckForEditUser(UserDTO userDTO)
+        {
+            return !DataBase.Users.GetAll()
+                                  .Where(u => u.Id != userDTO.Id)
+                                  .Any(u => u.UserName.Equals(userDTO.UserName) ||
+                                            u.Email.Equals(userDTO.Email));
```
Comment (AleksandroBoko) [inline StoreApp.BLL/Services/UserValidateService.cs]: 1. Just apply default formatting. 
2. Why have you applied such an approach? It seems to me you can use only **Any**

File: StoreApp/Controllers/AccountController.cs [hunk 3]
```diff
@@ -70,19 +88,194 @@ public ActionResult Login(UserLoginViewModel model)
...
+            if (Session["Role"] as string == "admin")
+            {
+                if (user.Role.Equals("0"))
+                {
+                    user.Role = "user";
+                }
+                else if (user.Role.Equals("1"))
+                {
+                    user.Role = "admin";
+                }
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/AccountController.cs]: I propose to create Enum with roles

File: StoreApp/Controllers/AccountController.cs [hunk 4]
```diff
@@ -70,19 +88,194 @@ public ActionResult Login(UserLoginViewModel model)
...
+
+                UserDTO userDTO = webMapper.config.Map<UserViewModel, UserDTO>(user);
+
+                userService.Edit(userDTO);
+
+                TempData["Message"] = user.UserName + " edit success";
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/AccountController.cs]: I propose to use this $"" to combine strings.
Also you should write readable message)

File: StoreApp/Controllers/AccountController.cs [hunk 5]
```diff
@@ -70,19 +88,194 @@ public ActionResult Login(UserLoginViewModel model)
...
+        [Authorize]
+        public ActionResult DeleteUser(UserViewModel user)
+        {
+            if (Session["Role"] as string == "admin")
+            {
+                if (user != null)
+                {
+                    userService.Delete(user.Id);
+
+                    TempData["Message"] = user.UserName + " deleted success";
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/AccountController.cs]: the same

File: StoreApp/Controllers/AccountController.cs [hunk 6]
```diff
@@ -70,19 +88,194 @@ public ActionResult Login(UserLoginViewModel model)
...
+                }
+                else
+                {
+                    TempData["Message"] = user.UserName + " don't deleted";
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/AccountController.cs]: the same

File: StoreApp/Views/Account/EditUserAdmin.cshtml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        </div>
+
+        <div class="form-group">
+            @Html.LabelFor(model => model.Role, htmlAttributes: new { @class = "control-label col-md-2" })
+            <div class="col-md-10">
+                @Html.DropDownListFor(model => model.Role, new List<SelectListItem>
+                                                               {
+                                                                   new SelectListItem { Value = "0", Text = "user" },
+                                                                   new SelectListItem { Value = "1", Text = "admin"} 
+                                                               },
```
Comment (AleksandroBoko) [inline StoreApp/Views/Account/EditUserAdmin.cshtml]: Try to create this list in controller not in view

File: StoreApp/Views/Account/DeleteUser.cshtml [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        <hr />
+        @Html.ValidationSummary(true, "", new { @class = "text-danger" })
+        <div class="form-group">
+            @Html.LabelFor(model => model.Email, htmlAttributes: new { @class = "control-label col-md-2" })
+            <div class="col-md-10">
+                @Html.EditorFor(model => model.Email, new { htmlAttributes = new { @class = "form-control", @readonly = "readonly" } })
+                @Html.ValidationMessageFor(model => model.Email, "", new { @class = "text-danger" })
+            </div>
+        </div>
+
```
Comment (AleksandroBoko) [inline StoreApp/Views/Account/DeleteUser.cshtml]: You don't need to add this empty space

File: StoreApp/Views/Account/DetailsUser.cshtml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+<h3>Details user @Model.UserName for admin</h3>
+
+<div>
+    <hr />
+    <dl class="dl-horizontal">
+        <dt>
+            @Html.DisplayNameFor(model => model.Email)
+        </dt>
+
```
Comment (AleksandroBoko) [inline StoreApp/Views/Account/DetailsUser.cshtml]: Remove empty space

File: StoreApp.BLL/Services/UserValidateService.cs (hunk 1 above)
Comment (Activict) [reply StoreApp.BLL/Services/UserValidateService.cs]: fixed

File: StoreApp.BLL/Services/UserValidateService.cs (hunk 2 above)
Comment (Activict) [reply StoreApp.BLL/Services/UserValidateService.cs]: fixed

File: StoreApp/Controllers/AccountController.cs (hunk 4 above)
Comment (Activict) [reply StoreApp/Controllers/AccountController.cs]: fixed

File: StoreApp/Controllers/AccountController.cs (hunk 5 above)
Comment (Activict) [reply StoreApp/Controllers/AccountController.cs]: fixed

File: StoreApp/Controllers/AccountController.cs (hunk 6 above)
Comment (Activict) [reply StoreApp/Controllers/AccountController.cs]: fixed

File: StoreApp/Views/Account/EditUserAdmin.cshtml (hunk 7 above)
Comment (Activict) [reply StoreApp/Views/Account/EditUserAdmin.cshtml]: fixed

File: StoreApp/Views/Account/DeleteUser.cshtml (hunk 8 above)
Comment (Activict) [reply StoreApp/Views/Account/DeleteUser.cshtml]: fixed

File: StoreApp/Views/Account/DetailsUser.cshtml (hunk 9 above)
Comment (Activict) [reply StoreApp/Views/Account/DetailsUser.cshtml]: fixed

File: StoreApp/Controllers/AccountController.cs (hunk 3 above)
Comment (Activict) [reply StoreApp/Controllers/AccountController.cs]: fixed

**Pull request merged**